### PR TITLE
Stop anti-sniper hate

### PR DESCRIPTION
**Pull request by TheVilest**
*In the arms of an angel fades into the foreground*

My name is Vile, and i'm here today to stand up for all of the neglected sniper rifles in the Awakened Worlds. For too long have these innocent firearms been subjected to unfair treatment at the hands of their oppressors. For one hundred nuyen a day you can support a sniper that only wants to put a single semi-automatic round through their opponents head at a distance. This outlandish prejudice that supports fully automatic 18-20D assault rifles and machine guns with vastly less dodge'able attacks with APDS (compared to, at most, 14S/14D at max non-dodge modifying attacks) must be stopped in its tracks if we're to move forward into a future where sniper rifles can join hand in hand with all of the other firearms of the world in harmonious tranquility of murder. Gone are the days are improperly calculated vision modifiers of weather and light conditions, and now are the days to embrace our single round "holy fragging hell these are way easier to dodge, lol" breathren in a united front against all of the NPC's we can throw our hard earned APDS rounds at.

Yours truly,
Vile

P.S. - Fight me.

### CODE REVIEW

File: src/handler.cpp [hunk 1]
```diff
@@ -1063,6 +1063,8 @@ void affect_total(struct char_data * ch)
     // Ranged weapons grant 0 reach if pistol, 2 reach if anything else with a bayonet, 1 reach for all other categories.
     else {
       if (GET_WEAPON_SKILL(weapon) != SKILL_PISTOLS)
```
Comment (luciensadi) [inline src/handler.cpp:1065]: Chaining the check like this technically works, but it's a lot cleaner to write `if (GWS != pistols && GWS != tasers &&...)`. More readable too.

File: src/newfight.hpp [hunk 2]
```diff
@@ -197,9 +197,11 @@ struct melee_combat_data {
     if (weapon) {
       if (ranged_combat_mode) {
         if (does_weapon_have_bayonet(weapon)) {
+          power += 2;
```
Comment (luciensadi) [inline src/newfight.hpp:200]: Why are we adding 2 power? Can you toss in a comment with the source reference?

**Review by luciensadi — CHANGES_REQUESTED** [review]

File: src/newfight.hpp (hunk 2 above)
Comment (TheVilest) [reply src/newfight.hpp]: Done. Rifle-butts do +2, pistol whipping is only +1, no differentiation between the two in code atm.

File: src/newfight.hpp [hunk 3]
```diff
@@ -197,9 +197,11 @@ struct melee_combat_data {
     if (weapon) {
       if (ranged_combat_mode) {
         if (does_weapon_have_bayonet(weapon)) {
+          power += 2; // Cannon Companion page 33 table, bayonet is listed as +2 Power (or errata if misprinted) -Vile
           dam_type = TYPE_PIERCE;
           skill = SKILL_POLE_ARMS;
         } else {
+          power += 2; // Cannon Companion page 11 table, rifle-butt is listed as +2 Power. -Vile
```
Comment (luciensadi) [inline src/newfight.hpp:204]: What about pistol butt etc? This applies to all ranged weapons if it's done here. If they're all supposed to have +2, that's fine.

Comment (TheVilest) [reply src/newfight.hpp:208]: Pistol is supposed to only be +1 in comparison

**Comment by luciensadi** [comment]
Thanks for this!

**Pull request merged**